examples/openshift-lightspeed.yaml

```diff
@@ -0,0 +1,161 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: openshift-lightspeed
+    app.kubernetes.io/component: lightspeed-w-rag
+    app.kubernetes.io/instance: lightspeed-w-rag
+    app.kubernetes.io/name: lightspeed-w-rag
+    app.kubernetes.io/part-of: lightspeed-w-rag-app
+  name: lightspeed-w-rag
+spec:
+  progressDeadlineSeconds: 600
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      app: openshift-lightspeed
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: openshift-lightspeed
+        deployment: lightspeed-w-rag
+    spec:
+      volumes:
+        - name: olsconfig
+          configMap:
+            name: olsconfig
+        - name: openai
+          secret:
+            secretName: openai
+      containers:
+      - image: quay.io/openshift/lightspeed-service-api-rag:latest
+        imagePullPolicy: Always
+        name: lightspeed-w-rag
+        env:
+          - name: OLS_CONFIG_FILE
+            value: /app-root/config/olsconfig.yaml
+        ports:
+        - containerPort: 8080
+          protocol: TCP
+        resources: {}
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+        volumeMounts:
+          - name: olsconfig
+            mountPath: /app-root/config
+          - name: openai
+            mountPath: /app-root/config/openai
+      dnsPolicy: ClusterFirst
+      restartPolicy: Always
+      schedulerName: default-scheduler
+      securityContext: {}
+      terminationGracePeriodSeconds: 30
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: openshift-lightspeed
+    app.kubernetes.io/component: lightspeed-w-rag
+    app.kubernetes.io/instance: lightspeed-w-rag
+    app.kubernetes.io/name: lightspeed-w-rag
+    app.kubernetes.io/part-of: lightspeed-w-rag-app
+  name: lightspeed-w-rag
+spec:
+  internalTrafficPolicy: Cluster
+  ipFamilies:
+  - IPv4
+  ipFamilyPolicy: SingleStack
+  ports:
+  - name: 8080-tcp
+    port: 8080
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: openshift-lightspeed
+    deployment: lightspeed-w-rag
+  sessionAffinity: None
+  type: ClusterIP
+
+---
+apiVersion: route.openshift.io/v1
+kind: Route
```

Comment:
until we have auth on by default we should probably not create a route.  This essentially creates a public facing openai proxy that uses our credentials/bill

the console won't need a route anyway

Reply:
How would you access OLS for testing without a route when it's running "over there" (in a cluster)?

Even if you are using OpenShift Local, you would still need a route.

Without a route, the only way to access OLS from outside the cluster for testing would be to spin up a pod inside the cluster that has curl and then use `oc rsh` or `oc exec` or similar.

I'm fine to remove the route, I just don't understand how you expect to access it -- can you explain?

Reply:
ah, forgot about port-forward. good call.